### PR TITLE
Rework messaging

### PR DESCRIPTION
**Pull request by tjjfvi**

### CODE REVIEW

File: packages/server/src/clientMessenger.ts [hunk 1]
```diff
@@ -1,42 +1,48 @@
...
   hashToUrl: (hash: Hash<unknown>) => string,
-  rendererMessenger: ServerRendererMessenger,
+  createRendererMessenger: () => Promise<ServerRendererMessenger>,
   bundlerMessenger?: ServerBundlerMessenger,
-) => {
+  serverEmitter: ServerEmitter,
+}) => {
+  let renderer = createRendererMessenger()
   const messenger: ServerClientMessenger = createMessenger({
-    async *ping(period){
```
Comment (vjjft) [inline packages/server/src/clientMessenger.ts:13]: no new implementation of `ping` (should this have been caught at type level?)

File: packages/messages/tests/createMessenger.test.ts [hunk 2]
```diff
@@ -1,55 +1,65 @@
...
+  const eventFn4 = jest.fn()
+  msgrA.on("heyWereHavingAReallyDumbPromiseResolveParty", eventFn1)
+  msgrB.on("heyWereHavingAReallyDumbPromiseResolveParty", eventFn2)
+  msgrB.once("heyWereHavingAReallyDumbPromiseResolveParty", eventFn3)
+  ;(async () => {
+    for await (const event of msgrA.on("heyWereHavingAReallyDumbPromiseResolveParty"))
+      eventFn4(event)
+  })()
+  expect(msgrA.once("heyWereHavingAReallyDumbPromiseResolveParty")).resolves.toStrictEqual([2, 10])
+  expect(await msgrA.reallyDumbPromiseResolve(2, 10)).toBe(2)
```
Comment (vjjft) [inline packages/messages/tests/createMessenger.test.ts]: consider more informative naming

File: packages/messages/tests/createMessenger.test.ts [hunk 3]
```diff
@@ -1,55 +1,65 @@
...
+
+type TestMessenger = Messenger<TestShape, TestShape, TestEvents>
+
+const createTestMessenger = (connection: Connection<unknown>) => {
+  const messenger: TestMessenger = createMessenger({
+    impl: {
+      promiseResolve: value => Promise.resolve(value),
+      async reallyDumbPromiseResolve(value, count){
+        messenger.emit("heyWereHavingAReallyDumbPromiseResolveParty", value, count)
+        if(count <= 0)
```
Comment (vjjft) [inline packages/messages/tests/createMessenger.test.ts]: `===`

File: packages/messages/tests/createMessenger.test.ts [hunk 4]
```diff
@@ -1,55 +1,65 @@
...
+  msgrA.on("heyWereHavingAReallyDumbPromiseResolveParty", eventFn1)
+  msgrB.on("heyWereHavingAReallyDumbPromiseResolveParty", eventFn2)
+  msgrB.once("heyWereHavingAReallyDumbPromiseResolveParty", eventFn3)
+  ;(async () => {
+    for await (const event of msgrA.on("heyWereHavingAReallyDumbPromiseResolveParty"))
+      eventFn4(event)
+  })()
+  expect(msgrA.once("heyWereHavingAReallyDumbPromiseResolveParty")).resolves.toStrictEqual([2, 10])
+  expect(await msgrA.reallyDumbPromiseResolve(2, 10)).toBe(2)
+  expect(eventFn1).toHaveBeenCalledTimes(11)
```
Comment (vjjft) [inline packages/messages/tests/createMessenger.test.ts]: `10`

**Review by vjjft — CHANGES_REQUESTED** [review]
@tjjfvi reminder to follow up on: https://github.com/tjjfvi/escad/blob/22938f9549c5189fab2a0ed38d62ab1a34f0f4c5/packages/messages/src/Connection.ts#L4

File: packages/server/src/clientMessenger.ts (hunk 1 above)
Comment (tjjfvi) [reply packages/server/src/clientMessenger.ts:13]: per discussion, the previous `ping` method was never called before, so I'm going to drop the `ping` event

**Pull request merged**